server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueModule.java

```diff
@@ -23,14 +23,19 @@
 import org.apache.james.queue.memory.MemoryMailQueueFactory;
 
 import com.google.inject.AbstractModule;
-import com.google.inject.Scopes;
+import com.google.inject.Provides;
+import com.google.inject.Singleton;
 
 public class MemoryMailQueueModule extends AbstractModule {
 
     @Override
     protected void configure() {
-        bind(MemoryMailQueueFactory.class).in(Scopes.SINGLETON);
 
-        bind(MailQueueFactory.class).to(MemoryMailQueueFactory.class);
+    }
+
+    @Provides
+    @Singleton
+    public MailQueueFactory<?> createActiveMailQueueFactory(MemoryMailQueueFactory memoryMailQueueFactory) {
```

Comment:
not needed, there's the famous TypeLiteral for that.

Reply:
I tend to prefer this syntax, and it is the one used for ActiveMQModule.

Reply:
We need this as the _MailQueueItemDecoratorFactory_ is bounded in another module (it's injected in the _MemoryMailQueueFactory_).